src/engine/BMInterfaceGame.php

```diff
@@ -465,15 +465,21 @@ public function save_join_game_decision($playerId, $gameId, $decision) {
 
         $playerIdx = array_search($playerId, $game->playerIdArray);
 
-        if (FALSE === $playerIdx) {
+        if ($playerIdx !== FALSE) {
+            $player = $game->playerArray[$playerIdx];
+            $player->waitingOnAction = FALSE;
+            $decisionFlag = ('accept' == $decision);
+            $player->hasPlayerAcceptedGame = $decisionFlag;
+        } elseif (($playerId === $game->creatorId) &&
```

Comment:
I threw in a bunch of debug-print statements, and i think this is your problem.  `$playerId` is a string and `$game->creatorId` is an int. 

Assuming i'm right:
* I have no idea why it works for you (maybe it's a browser thing, though?)
* Ironically, this is exactly the problem i was trying to solve with #2613, but even if we haven't merged that into this dev site (and it sounds like we should, just to be on the safe side), it won't save you, because playerId is coming from `$_SESSION['user_id']`, and that's the one thing we *don't* have a consistent story around casting.

My takeaway is that we gotta do the part of #712 where we tackle the session variables and cast them consistently, so that we're not counting on individual functions in ApiResponder to remember to cast `$_SESSION['user_id']`.